3rdparty/CMakeLists.txt

```diff
@@ -111,6 +111,7 @@ set(THIRD_PARTY_INCLUDE ${THIRD_PARTY_INCLUDE} PARENT_SCOPE)
 add_library(lfs SHARED
   ${CMAKE_CURRENT_SOURCE_DIR}/lua/luafilesystem/src/lfs.c
 )
+target_link_libraries(lfs PRIVATE libujit_shared)
```

Comment:
This looks troublesome to me...
1) Why should 3rd-party library link to our library?
2) If BUILD_SHARED_LIBS is OFF, lfs will be a static library. Linking to shared library is asking for trouble then...

Reply:
> Why should 3rd-party library link to our library?
Because it needs some Lua engine (it's a Lua/C library)

> If BUILD_SHARED_LIBS is OFF, lfs will be a static library. Linking to shared library is asking for trouble then...

Support for `BUILD_SHARED_LIBS` is a different thing, but I'll clean up CMake as discussed off-tracker.